modules/siddhi-core/src/main/java/io/siddhi/core/stream/input/InputManager.java

```diff
@@ -69,6 +74,20 @@ public InputHandler getInputHandler(String streamId) {
         }
     }
 
+    public TableInputHandler getTableInputHandler(String tableId) {
+        TableInputHandler tableInputHandler = tableInputHandlerMap.get(tableId);
+        if (tableInputHandler == null) {
+            synchronized (this) {
+                tableInputHandler = tableInputHandlerMap.get(tableId);
+                if (tableInputHandler == null) {
+                    return constructTableInputHandler(tableId);
+                }
+                return tableInputHandler;
+            }
+        }
+        return tableInputHandler;
+    }
+
```

Comment:
It seems we are doing the same thing (`tableInputHandlerMap.get(tableId);`) twice. Shall we make this as follows:
```
public TableInputHandler getTableInputHandler(String tableId) {
        synchronized (this) {
            TableInputHandler tableInputHandler = tableInputHandlerMap.get(tableId);
            if (tableInputHandler == null) {
                  return constructTableInputHandler(tableId);
            }
            return tableInputHandler;
        }
}
```

Reply:
Fixed in 49a028a